src/shrub/src/grade/grade_core.php

```diff
@@ -28,14 +28,16 @@ function grade_GetByNodeAuthor( $node_id, $author_id ) {
 
 
 function grade_GetByAuthorParent( $author_id, $parent_id ) {
-	return db_QueryFetchIdKeyValue(
+	return db_QueryFetchIdKeyValueMeta(
 		"SELECT
 			node AS id,
 			name,
-			value
+			value,
+			timestamp
 		FROM ".SH_TABLE_PREFIX.SH_TABLE_GRADE."
 		WHERE author=? AND parent=?
 		;",
+		'-timestamp',
 		$author_id,
 		$parent_id
 	);
```

Comment:
If you want different output, call a different function, and/or transform it here.

Reply:
Okay, not exactly what I meant (though technically correct). 

What I more meant was a different `db_QueryFetch` call should be used.

`db_QueryFetch` will give you an array of data something like this:

```js
[
    {
        'id': 54,
        'name': "geoffry",
        'value': 77,
        'timestamp': "2008-11-11T00:00:00Z"
    }
]
```

And `db_QueryFetchWithIntKey('id', ...)` will give you something like this:

```js
{
   '54': {
        'id': 54,
        'name': "geoffry",
        'value': 77,
        'timestamp': "2008-11-11T00:00:00Z"
    }
}
```